### PR TITLE
Query cloud specific env vars in task setup

### PR DESCRIPTION
**Pull request by hemildesai**
Adds a class method to `Cloud` called `query_env_vars` which can query cloud specific env vars. Addresses https://github.com/skypilot-org/skypilot/issues/2287

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
    - [x] Ran `sky launch -- printenv` and verified that `KUBERNETES_SERVICE_PORT` etc are included
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/kubernetes.py [hunk 1]
```diff
@@ -406,3 +406,27 @@ def query_status(cls, name: str, tag_filters: Dict[str, str],
...
         return cluster_status
+
+    @classmethod
+    def query_env_vars(cls, name: str) -> Dict[str, str]:
+        namespace = kubernetes_utils.get_current_kube_config_context_namespace()
+        pod = kubernetes.core_api().list_namespaced_pod(
+            namespace,
+            label_selector=f'skypilot-cluster={name},ray-node-type=head'
+        ).items[0]
+        response = kubernetes.stream()(
```
Comment (romilbhardwaj) [inline sky/clouds/kubernetes.py:417]: Is it possible to set a timeout on this stream call (like we have `_request_timeout=kubernetes.API_TIMEOUT` in the rest of our code)?

**Review by romilbhardwaj — COMMENTED** [review]
This is super timely for #2328, which needs access to envvars set by GKE to setup GPU access. Thanks for the great work @hemildesai! 

Tested it, works nicely with `sky launch`. Left some comments about some critical edge cases.

File: sky/backends/cloud_vm_ray_backend.py [hunk 2]
```diff
@@ -2828,6 +2828,11 @@ def _setup(self, handle: CloudVmRayResourceHandle, task: task_lib.Task,
         style = colorama.Style
         fore = colorama.Fore
 
+        if isinstance(handle.launched_resources.cloud, clouds.Kubernetes):
+            cloud_env_vars = handle.launched_resources.cloud.query_env_vars(
+                handle.cluster_name)
+            task.update_envs(cloud_env_vars)
```
[the remark this answers is not in the document]
Comment (hemildesai) [reply sky/backends/cloud_vm_ray_backend.py:2834]: That sounds good, I will update to give precedence to user defined env vars.

File: sky/clouds/kubernetes.py (hunk 1 above)
Comment (hemildesai) [reply sky/clouds/kubernetes.py:417]: Yes it should be. I'll add the timeout.

File: sky/backends/cloud_vm_ray_backend.py [hunk 3]
```diff
@@ -2828,6 +2828,11 @@ def _setup(self, handle: CloudVmRayResourceHandle, task: task_lib.Task,
         style = colorama.Style
         fore = colorama.Fore
 
+        if isinstance(handle.launched_resources.cloud, clouds.Kubernetes):
```
[the remark this answers is not in the document]
Comment (hemildesai) [reply sky/backends/cloud_vm_ray_backend.py:2831]: I can create a utility to update env vars and call it in both `_setup` and `_execute`

**Comment by romilbhardwaj** [comment]
Thanks @hemildesai! In addition to envs in YAML, I also tested inline envs with `sky exec test --env LIBRARY_PATH=/my/cuda/path -- printenv`. Works nicely, merging to `k8s_cloud_beta1`.

**Pull request merged**